src/integrations/admin/check-required-version.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Yoast\WP\SEO\Integrations\Admin;
+
+use WP_Error;
+use Yoast\WP\SEO\Conditionals\No_Conditionals;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+
+/**
+ * The Check_Required_Version class.
+ *
+ * This class checks if the required version of Yoast SEO is installed.
+ * It also adds the `Requires Yoast SEO` header to the list of headers and updates the comparison table for the plugin overwrite.
+ */
+class Check_Required_Version implements Integration_Interface {
+
+	use No_Conditionals;
+
+	/**
+	 * Initializes the integration.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'upgrader_source_selection', [ $this, 'check_required_version' ] );
+		\add_filter( 'install_plugin_overwrite_comparison', [ $this, 'update_comparison_table' ], 10, 3 );
+	}
+
+	/**
+	 * Checks if the required version of Yoast SEO is installed.
+	 *
+	 * The code is partly inspired by Plugin_Upgrader::check_package() in wp-admin/includes/class-plugin-upgrader.php.
+	 *
+	 * @param string $source File source location.
+	 *
+	 * @return string|WP_Error The source location or a WP_Error object if the required version is not installed.
+	 */
+	public function check_required_version( $source ) {
+		global $wp_filesystem;
+
+		$info = [];
+
+		if ( \is_wp_error( $source ) ) {
+			return $source;
+		}
+
+		$working_directory = \str_replace( $wp_filesystem->wp_content_dir(), \trailingslashit( \WP_CONTENT_DIR ), $source );
+		if ( ! \is_dir( $working_directory ) ) { // Confidence check, if the above fails, let's not prevent installation.
+			return $source;
+		}
+
+		// Check that the folder contains at least 1 valid plugin.
+		$files = \glob( $working_directory . '*.php' );
+		if ( $files ) {
+			foreach ( $files as $file ) {
+				$info = \get_plugin_data( $file, false, false );
+				if ( ! empty( $info['Name'] ) ) {
+					break;
+				}
+			}
+		}
```

Comment:
I have checked on a test site with the 14 most used plugins: running a mass upgrade of them was not visibly affected by the code above.

Reply:
As per [our discussion after that investigation](https://yoast.slack.com/archives/C03KU0EHCNQ/p1724412487478839), we decided to move forward with this, since it's a negligible difference :)